src/init.cpp

```diff
@@ -1031,6 +1031,19 @@ bool AppInitParameterInteraction(const ArgsManager& args, bool use_syscall_sandb
         return InitError(_("No proxy server specified. Use -proxy=<ip> or -proxy=<ip:port>."));
     }
 
+    if (args.IsArgSet("-reindex-chainstate")) {
+        // indexes that must be deactivated to prevent index corruption, see #24630
+        if (args.GetBoolArg("-coinstatsindex", DEFAULT_COINSTATSINDEX)) {
+            return InitError(_("reindex-chainstate with active coinstatsindex is not possible. Please deactivate the index temporarily or perform a full -reindex, which will also rebuild the index."));
```

Comment:
In commit "init: disallow reindex-chainstate with optional indexes" (93fa8f651242af5354ba92f7aaec18e878d4f840)

Minor: Here and below I think you could make recommendations more explicit. I would say "-reindex-chainstate option is not compatible with -coinstatsindex. Please temporarily disable -coinstatsindex while using -reindex-chainstate, or replace -reindex-chainstate with -reindex to fully rebuild all indexes." But current wording is also OK if you prefer it.

Reply:
Thanks, I took your suggestion (just didn't use a "-" after "disable" because here the index is meant, not the command option).